patches/server/0984-SculkCatalyst-bloom-API.patch

```diff
@@ -0,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Oliwier Miodun <naczs@blueflow.pl>
+Date: Mon, 10 Jul 2023 17:59:42 +0200
+Subject: [PATCH] SculkCatalyst bloom API
+
+== AT ==
+public net.minecraft.world.level.block.entity.SculkCatalystBlockEntity$CatalystListener bloom(Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/util/RandomSource;)V
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java b/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
+index dc7d6c443af206f50d2b4ca260d64bb86b67553c..9100410f639675a882cf58b817ee61d6638c33e2 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftSculkCatalyst.java
+@@ -9,4 +9,22 @@ public class CraftSculkCatalyst extends CraftBlockEntityState<SculkCatalystBlock
+     public CraftSculkCatalyst(World world, SculkCatalystBlockEntity tileEntity) {
+         super(world, tileEntity);
+     }
++
++    // Paper start - SculkCatalyst bloom API
++    @Override
++    public void bloom(@org.jetbrains.annotations.NotNull io.papermc.paper.math.Position position, int charge) {
++        com.google.common.base.Preconditions.checkNotNull(position);
++        requirePlaced();
++
++        getTileEntity().getListener().bloom(
++            world.getHandle(),
++            getTileEntity().getBlockPos(),
++            getTileEntity().getBlockState(),
++            world.getHandle().getRandom()
++        );
++        getTileEntity().getListener().getSculkSpreader().addCursors(
++            new net.minecraft.core.BlockPos(position.blockX(), (int) (position.blockY() + 0.5D), position.blockZ()),
```

Comment:
Should we be doing this position mutation with the API method? I feel like its to offset from the emitter position within the game, but the API can specify whatever position it wants.

Reply:
The +0.5 is copied directly from game's code. I assume it's for cases like the entity standing on e.g. a slab, if someone just passes an entity's Position imho should still add that half.

Reply:
Or we could just have an 'overload' for LivingEntity